app/uk/gov/hmrc/cgtpropertydisposals/controllers/SubscriptionController.scala

```diff
@@ -72,14 +88,34 @@ class SubscriptionController @Inject()(
         case BackendError(e) =>
           logger.warn("Error while trying to subscribe", e)
           InternalServerError
-        case EnrolmentError(cgtReference) =>
-          logger.warn("Error while trying to allocate enrolment")
-          Ok(Json.toJson(SubscriptionResponse(cgtReference))) // We return a 200 OK even though it is an error as we will retry asynchronously
+        case FailedTaxEnrolmentCall(taxEnrolmentRequest, cgt) =>
+          logger.warn("Error while trying to allocate enrolment - sending message to retry asynchronously")
+          retry(taxEnrolmentRequest).value.map {
+            case Left(error) =>
+              logger.warn(s"Failed to submit async retry request for $taxEnrolmentRequest: error [$error]")
+            case Right(_) => logger.info(s"Successfully submitted async retry request for $taxEnrolmentRequest")
+          }
+          Ok(Json.toJson(SubscriptionResponse(cgt)))
       }, { r =>
         Ok(Json.toJson(r))
       }
     )
   }
+
+  private def retry(taxEnrolmentRequest: TaxEnrolmentRequest): EitherT[Future, Error, Unit] =
+    EitherT[Future, Error, Unit](
+      Future {
+        Right(
+          taxEnrolmentRetryProvider.taxEnrolmentRetryManager
+            .tell(
```

Comment:
if the message fails to reach the actor we're in a bit of trouble here since the enrolment will never get retried. This should probably be an ask so that you can handle an `AskTimeoutException`.

Reply:
We discussed this last Thursday. Because each request is inserted into the database before the tax enrolment request is made we don't lose information. On system recovery, the request can always be replayed. I can change it to an ask if you prefer.